py3status/py3.py

```diff
@@ -679,3 +740,67 @@ def threshold_get_color(self, value, name=None):
         setattr(self._py3status_module, color_name, color)
 
         return color
+
+    def _get_url(self, url):
+        """
+        Get the HTTPResponse for the HTTP GET request to `url`.
+
+        Raises `Py3.URLError` on Connection Errors.
+        Raises `Py3.ClientError` on HTTP 4xx Errors.
+        Raises `Py3.ServerError` on HTTP 5xx Errors.
+        """
+        try:
+            response = HTTPResponse(urlopen(url))
```

Comment:
I did it partly, because I had the mindset of working with `HTTPRequest` and `HTTPResponse` objects.  When building the `HTTPRequest` object, the URL is defined there, along with other attributes.  It may do some processing prior to making the full request, which would return the `HTTPResponse` object.

Now, we aren't implementing the full feature set, but `urlopen(url)` is, in a sense, taking the place of `http.request()`, since it returns a `Response` object.